crates/biome_css_analyze/src/lint/nursery/no_irregular_whitespace_css.rs

```diff
@@ -0,0 +1,105 @@
+use biome_analyze::{
+    context::RuleContext, declare_lint_rule, Ast, Rule, RuleDiagnostic, RuleSource,
+};
+use biome_console::markup;
+use biome_css_syntax::{AnyCssRule, CssLanguage};
+use biome_rowan::{AstNode, Direction, SyntaxToken, TextRange};
+
+const IRREGULAR_WHITESPACES: &[char; 22] = &[
+    '\u{c}', '\u{b}', '\u{85}', '\u{feff}', '\u{a0}', '\u{1680}', '\u{180e}', '\u{2000}',
+    '\u{2001}', '\u{2002}', '\u{2003}', '\u{2004}', '\u{2005}', '\u{2006}', '\u{2007}', '\u{2008}',
+    '\u{2009}', '\u{200a}', '\u{200b}', '\u{202f}', '\u{205f}', '\u{3000}',
+];
+
+declare_lint_rule! {
+    /// Disallows the use of irregular whitespace.
+    ///
+    /// Using irregular whitespace would lead to the failure of selecting the correct target.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```css,expect_diagnostic
+    /// .firstClass.secondClass {
+    ///   color: red;
+    /// }
+    /// ```
+    ///
+    /// ```css,expect_diagnostic
+    /// .firstClass .secondClass {
+    ///   color:red;
+    /// }
+    /// ```
+    /// ### Valid
+    ///
+    /// ```css
+    /// .firstClass .secondClass {
+    ///   color: red;
+    /// }
+    /// ```
+    ///
+    pub NoIrregularWhitespaceCss {
+        version: "1.8.0",
+        name: "noIrregularWhitespaceCss",
+        language: "css",
+        recommended: false,
```

Comment:
```suggestion
        recommended: true,
```

Reply:
I don't think we should recommend it. It's a slow rule, and a niche one.